pkg/webhook/admission/pod/storage_initializer_injector_test.go

```diff
@@ -2723,6 +2723,13 @@ func TestInjectModelcar(t *testing.T) {
 			t.Errorf("Expected two containers but got %d", len(pod.Spec.Containers))
 		}
 
+		// Check that an init container has been injected, and it is the model container
+		if len(pod.Spec.InitContainers) != 1 {
+			t.Errorf("Expected one init container but got %d", len(pod.Spec.InitContainers))
+		} else if pod.Spec.InitContainers[0].Name != ModelcarInitContainerName {
+			t.Errorf("Expected the init container to be the model but got %s", pod.Spec.InitContainers[0].Name)
+		}
+
```

Comment:
You might want to add some more checks on the structure of the init-container, e.g. that both requests & limits are set, and that the commands contains some key aspects (I rarely check on literal strings for equality, but e.g. that the messaging contains important keywords that might help the user to troubleshoot, too.

Reply:
OK. I added the additional checks.